iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/helpers/X509CertificateGenerator.java

```diff
@@ -0,0 +1,164 @@
+/*
+ *  Copyright (c) Microsoft. All rights reserved.
+ *  Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+package com.microsoft.azure.sdk.iot.common.helpers;
+
+import com.microsoft.azure.sdk.iot.deps.util.Base64;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.cert.X509CertificateHolder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+
+import java.math.BigInteger;
+import java.security.*;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Date;
+
+/**
+ * Utility class for generating self signed certificates, and computing their thumprints
+ */
+public class X509CertificateGenerator
```

Comment:
remove the original X509Cert generation class

Reply:
I want to leave that class for now, since it has code for generating intermediate certificates, whereas this class only has a root cert. There will probably be a point later where we have tests for those intermediate cert scenarios, and I'd like to still have that code present when it happens